src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java

```diff
@@ -642,16 +644,29 @@ private void tryUnstuck()
             {
                 // Unstuck with path
                 final List<BlockPos> checkPositions = new ArrayList<>();
-                final PathPoint next = path.getPathPointFromIndex(path.getCurrentPathIndex());
+                PathPoint next = path.getPathPointFromIndex(Math.min(path.getCurrentPathIndex() + 1, path.getCurrentPathLength() - 1));
 
                 // Blocks in front of the worker
-                for (int i = 0; i <= 2; i++)
+                for (int i = 0; i < 2; i++)
                 {
                     checkPositions.add(new BlockPos(next.x, next.y + i, next.z));
                 }
-
-                // Block above the worker
-                checkPositions.add(new BlockPos(worker.getPosition().getX(), worker.getPosition().getY() + 2, worker.getPosition().getZ()));
+                if (next.previous != null)
```

Comment:
What does this change do?

Reply:
Lookup more than one block ahead, fixes some really weird edges cases

Reply:
actually previous should be the node behind, not the one ahead

Reply:
it's node behind, this while iterates in reverse order (ie. from end to start)